### DISCUSSION TITLE
Allow lambda attributes in expression trees

### DISCUSSION THREAD

**Discussion by FLAMESpl** [dotnet/csharplang#7247] [Language Ideas] [answered]
In csharp 10 we have allowed for declaring attributes on parameters of lambda methods as such:

```csharp
Func<string, string> func = ([Required] str) => str.ToLower();
```

however if it is not allowed in expression trees:

```csharp
Expression<Func<string, string>> func1 = ([Required] str) => str.ToLower();
```

What reason is behind this? I can see handly applications. In my case I am working on workflows engine where I capture lambda expression trees, walk through them to extract some metadata then serialize them and save in database. Currently I cannot inject any custom metadata into my expressions at the point where they are declared, but attributes would allow for this.

**Comment by TahirAhmadov** [comment, marked as answer]
This probably falls under the rubric of expanding expression trees: https://github.com/dotnet/csharplang/discussions/4727

**Reply by FLAMESpl** [reply]
I am excited to see expression tree's evolution
